packages/eslint-plugin-patternfly-react/lib/config/recommended.js

```diff
@@ -47,7 +47,9 @@ module.exports = {
     'react/jsx-filename-extension': 'off',
     'react/jsx-uses-vars': 'error',
     'react/no-danger': 'off',
-    'react/sort-comp': 'off'
+    'react/sort-comp': 'off',
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'warn'
```

Comment:
I don't believe we should be adding rules that we don't use ourselves in patternfly-react, even if it's good practice for others. How would folks disable the rule if they have trouble with it down the road?

Reply:
@dlabrecq I think the idea was that since `eslint-plugin-patternfly-react` is used in Foreman and most of its plugins, it would be better to just add the rule here in one place rather than every individual repo.

Like any eslint rule, these can be disabled with an `eslint-disable` or `eslint-disable-next-line` magic comment, or I think you can probably override rules in your own `.eslintrc` file.

Reply:
Sounds reasonable, but a little odd considering PatternFly does not use hooks. Can you please create an issue?

Reply:
//cc @dgutride

Reply:
@dlabrecq Ok, created #3092

Reply:
@dlabrecq patternfly-react are not going to use hooks in the future?

Reply:
Anything is possible, but we don't yet have any plans to introduce it. 

A benefit of using hooks is you don't need to write a class. Considering the majority of PF4 components don't maintain state, we only have a few component classes (e.g., tabs, dropdown, etc.). I don't know if hooks would be helpful enough to refactor those few components? 

I believe the PF team wants to discuss this a bit more...